app/controllers/secrets_controller.rb

```diff
@@ -66,8 +66,20 @@ def batch
       unless (secret = variable.last_secret)
         raise Exceptions::RecordNotFound, variable.resource_id
       end
-      
-      result[variable.resource_id] = secret.value
+
+      mime_type = variable.annotation('conjur/mime_type') || nil
+
+      if mime_type == 'application/octet-stream'
+        unless result.key?('application/octet-stream')
+          result['application/octet-stream'] = { 'encoding': 'base64' }
+        end
+        result['application/octet-stream'][variable.resource_id] = Base64.encode64(secret.value)
+      elsif secret.value.class == String && !secret.value.force_encoding('utf-8').valid_encoding?
```

Comment:
Here are some [more idiomatic ways](https://stackoverflow.com/questions/7749416/check-whether-a-variable-is-a-string-in-ruby) to determine if something is a String.

Reply:
I'd also add to John's comment that the whole block would read better if we remove the nested if/unless checks, and pre-assign the conditional clauses to meaningful variable names.

I'd also re-structure it to use [line of sight](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88).